app/library/contentList.tsx

```diff
@@ -0,0 +1,105 @@
+import axios from 'axios';
+import { Subject, Grade, Chapter, Resource, Topic } from '../types'
+
+const url = process.env.NEXT_PUBLIC_AF_DB_SERVICE_URL;
+
+export const getSubjects = async (subjectName: string): Promise<Subject[]> => {
+  try {
+    const response = await axios.get(`${url}/subject`, {
+      params: { name: subjectName },
+    });
+    return response.data;
+  } catch (error) {
+    console.error("Error in getSubjects:", error);
+    throw error;
+  }
+};
+
+export const getGrades = async (number: number): Promise<Grade[]> => {
+  try {
+    const response = await axios.get(`${url}/grade`, {
+      params: { number: number },
+    });
+    return response.data;
+  } catch (error) {
+    console.error("Error in getSubjects:", error);
+    throw error;
+  }
+};
+
+export const getChapters = async (subjectId?: number, gradeId?: number, limit?: number, offset?: number, id?: number): Promise<Chapter[]> => {
+  try {
+    const response = await axios.get(`${url}/chapter`, {
+      params: { id: id, subject_id: subjectId, grade_id: gradeId, limit, offset },
+    });
+    return response.data;
+  } catch (error) {
+    console.error("Error in getChapters:", error);
+    throw error;
+  }
+};
+
+export const getTopics = async (chapterIds: number[], limit: number, offset: number): Promise<Topic[]> => {
+  const topicPromises = chapterIds.map(async (chapterId) => {
+    try {
+      const response = await axios.get(`${url}/topic`, {
+        params: { chapter_id: chapterId, limit, offset },
+      });
+      return response.data || [];
+    } catch (error) {
+      console.error("Error in getTopics for chapterId", chapterId, ":", error);
+      return [];
+    }
+  });
+
+  const topicResponses = await Promise.all(topicPromises);
+  return topicResponses.flat();
+};
+
+export const getSource = async (sourceId: number) => {
+  try {
+    const response = await axios.get(`${url}/source`, {
+      params: { id: sourceId },
+    });
+
+    if (response.data) {
+      return response.data;
+    }
+  } catch (error) {
+    console.error("Error in getSource for sourceId", sourceId, ":", error);
+  }
+  return null;
+};
+
+export const getResourcesWithSource = async (topicIds: number[]): Promise<Resource[]> => {
```

Comment:
can you add a comment of how ResourcesWithSource looks like? for each topic ID, we figure out a chapters, and for each chapter we figure out source links?

Reply:
First we are figuring out subject here (default value is Physics)
Then we are fetching the chapter belonging to that paricular chapter and grade.
Next we are fetching topics belonging to a chapter.
And finally we are fetching resources belonging to a topic. Now since resources and sources and maintained in a different table, so I've made a func getResourcesWithSource which will be having an additional key named `link` where we are fetching source of the resource.